go-controller/pkg/ovn/obj_retry_master.go

```diff
@@ -493,10 +493,8 @@ func (h *masterEventHandler) AddResource(obj interface{}, fromRetryLoop bool) er
 			h.oc.setNodeEgressReady(node.Name, true)
 		}
 		isReachable := h.oc.isEgressNodeReachable(node)
-		if isReachable {
-			h.oc.setNodeEgressReachable(node.Name, true)
-		}
 		if hasEgressLabel && isReachable && isReady {
+			h.oc.setNodeEgressReachable(node.Name, isReachable)
```

Comment:
moving this here and in other places where isReachable && isReady is true makes sense to me. Do you think there is any scope for adding a test to catch this? some clever unit test that plays with these node fields?